core/unit_test/incremental/Test03b_Reduction.hpp

```diff
@@ -0,0 +1,159 @@
+//
+// Questions? Contact Christian R. Trott (crtrott@sandia.gov)
+//
+// ************************************************************************
+//@HEADER
+*/
+
+#include <Kokkos_Core.hpp>
+#include <gtest/gtest.h>
+
+/// @Kokkos_Feature_Level_Required:3
+// Unit test for reduction of native data type.
+// Assigns an index based value to elements of an array.
+// Performs an reduction over the addition operation.
+
+namespace Test {
+
+using value_type       = double;
+const double value     = 0.5;
+const int num_elements = 10;
+
+struct ReduceFunctor {
+  value_type *_data;
+
+  ReduceFunctor(value_type *data) : _data(data) {}
+
+  KOKKOS_INLINE_FUNCTION
+  void operator()(const int i, double &UpdateSum) const {
+    _data[i] = (i + 1) * value;
+    UpdateSum += _data[i];
+  }
+};
+
+template <class ExecSpace>
+struct TestReduction {
+  // Memory space type for Device and Host data
+  using d_memspace_type = typename ExecSpace::memory_space;
+  using h_memspace_type = Kokkos::HostSpace;
+
+  value_type *deviceData, *hostData;
+  value_type sum = 0.0;
+
+  // compare and equal
+  void check_correctness(double sum) {
+    int sum_local = 0;
+    for (int i = 0; i < num_elements; ++i) sum_local += (i + 1);
+
+    ASSERT_EQ(sum, sum_local * value)
+        << "The reduced value does not match the expected answer";
+  }
+
+  // Routine to allocate memory in a specific memory space.
+  template <class MemSpace>
+  value_type *allocate_mem(int N) {
+    return (static_cast<value_type *>(
+        Kokkos::kokkos_malloc<MemSpace>("deviceData", N * sizeof(value_type))));
+  }
+
+  // Routine to free the memory from a specific memory space.
+  template <class MemSpace>
+  void free_mem(value_type *data) {
+    Kokkos::kokkos_free<MemSpace>(data);
+  }
+
+  // Free the allocated memory
+  void free_mem() {
+    Kokkos::kokkos_free<d_memspace_type>(deviceData);
+    Kokkos::kokkos_free<h_memspace_type>(hostData);
+  }
+
+  // Allocate Memory for both device and host memory spaces
+  void init() {
+    // Allocate memory on Device space.
+    deviceData = allocate_mem<d_memspace_type>(num_elements);
+    ASSERT_NE(deviceData, nullptr);
+
+    // Allocate memory on Host space.
+    hostData = allocate_mem<h_memspace_type>(num_elements);
+    ASSERT_NE(hostData, nullptr);
+
+    // Initialize the sum value to zero.
+    sum = 0.0;
+  }
+
+  void check_correctness_and_cleanup() {
+    // Check if reduction has produced correct results
+    check_correctness(sum);
+
+    check_correctness_and_cleanup();
+
+    // free the allocated memory
+    free_mem<d_memspace_type>(deviceData);
+    free_mem<h_memspace_type>(hostData);
+  }
+
+  void sum_reduction() {
+    // Allocates memory for num_elements number of value_type elements in the
+    // host and device memory spaces.
+    init();
+
+    // Creates a range policy that uses static schedule for Cuda and dynamic
+    // schedule for other memory spaces.
+#if defined(KOKKOS_ENABLE_CUDA)
```

Comment:
Where is this coming from? I thought I had fixed this?

Reply:
Was a different file I guess. Anyway, there is no reason to do this here. `Kokkos::Dynamic` is totally fine to request from CUDA too.

Reply:
Modified the test to use Kokkos::Dynamic always.